### PR TITLE
Update README's roadmap and add more info about the trace/memory comparisons

### PR DESCRIPTION
**Pull request by jrchatruc**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,53 +104,45 @@ The milestone includes:
...
+- ✅ Builtins. Add the `BuiltinRunner` logic, then implement each builtin:
+  - ✅ `Range check (RC)`
+  - ✅ `Output`
+  - ✅ `Bitwise`
+  - ✅ `Ec_op`
+  - ✅ `Pedersen`
+  - ✅ `Keccak`
+  - ✅ `Poseidon`
+  - ✅ `Signature`
+  - ✅ `Segment Arena`
```
Comment (fmoletta) [inline README.md]:
```suggestion
  - [ ] `Segment Arena`
```
fake news

File: README.md [hunk 2]
```diff
@@ -90,53 +104,45 @@ The milestone includes:
...
+- ✅ Memory layouts. This is related to builtins but we will do it after implementing them.
```
Comment (fmoletta) [inline README.md]:
```suggestion
- ✅ Memory layouts. (plain, small, all_cairo)
```

File: README.md (hunk 1 above)
Comment (fmoletta) [reply README.md]: Or just remove it, as it is talked about below

File: README.md [hunk 3]
```diff
@@ -90,53 +104,45 @@ The milestone includes:
...
+- Hints. 
+    - ✅ Add the `HintProcessor` logic
+    - ✅ [Parsing of references](https://github.com/lambdaclass/cairo-vm/tree/main/docs/references_parsing)
+    - Implement each hint. This is a long task which we divided into milestones. You can check out their progress [here](https://github.com/lambdaclass/cairo-vm_in_go/milestones).
+- ✅ Proof mode. It's important to explain in detail what this is when we do it. It's one of the most obscure parts of the VM in my experience.
+- ✅ Program tests from Cairo VM in Rust.
+- Print debug information on failure (i.e. introduce the concept of a `VmException`)
```
Comment (fmoletta) [inline README.md:130]:
```suggestion
- Support for Temporary Memory
    - Handle temporary addresses in memory-related code
    - Implement relocation process for temporary memory
- Print debug information on failure (i.e. introduce the concept of a `VmException`)
```

File: README.md (hunk 1 above)
Comment (jrchatruc) [reply README.md]: Removed here [05ccbf8](https://github.com/lambdaclass/cairo-vm_in_go/pull/270/commits/05ccbf8f60cde0d273e14efdd6f1121fa582487c)

**Review by fmoletta — APPROVED** [review]

**Review by entropidelic — APPROVED** [review]

**Pull request merged**